packages/langium/src/grammar/grammar-util.ts

```diff
@@ -55,19 +56,26 @@ function isDataTypeRuleInternal(rule: ast.ParserRule, visited: Set<ast.ParserRul
     return true;
 }
 
-export function findNameAssignment(rule: ast.ParserRule): ast.Assignment | undefined {
-    for (const node of streamAllContents(rule)) {
+export function findNameAssignment(type: ast.AbstractType): ast.Assignment | undefined {
```

Comment:
Calling this function during validation leads to a stack overflow. We need to deal with recursive rulecalls similar to how you dealt with it in the `isDataTypeRule` function.